src/Sarif/Visitors/VersionOneLogicalLocationKeyToFullyQualifiedNameMappingVisitor.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Sarif.VersionOne;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    /// <summary>
+    /// This class visits all locations in a SARIF v1 log file. On observing use of result.logicalLocationKey,
+    /// which is used to disambiguate logical locations in the result.logicalLocations dictionary that have a 
+    /// common fully qualified name (but which are different types), the visitor creates a mapping between
+    /// the logical location key and its associated fully qualified name. This allows the v2 transformation
+    /// in particular to more easily populate its logical location equivalents. Additionally, the visitor 
+    /// stores a mapping from fully qualified name to decorated name, if one exists. These data (the
```

Comment:
>exists [](start = 77, length = 6)

This is the thing I worried about in `SarifVersionOneToCurrentVisitor.cs`. I would expect a mapping from key to decorated name, not from FQLN to decorated name, since there could be two logical locations with the same FQLN but different decorated names. In fact, come to think of it, one of the test files has this situation.

Reply:
Very nice catch. The PREfast samples have this condition. I believe all these samples, however, have been converted to pre-release v2. I will add a test for this case.

---
In reply to: [244240540](https://github.com/Microsoft/sarif-sdk/pull/1185#discussion_r244240540) [](ancestors = 244240540)